### ISSUE TITLE
Disable no-unused-vars with variables returning from "reactive" and "ref" or methods

### ISSUE THREAD

**Issue by ndr47** [vuejs/eslint-plugin-vue#1153]
**Please describe what the rule should do:**
Disable "no-unused-vars" rules when using Composition API for variables set by _reactive_ and _ref_.
Also for _methods_ / _computed_ 

**What category should the rule belong to?**
- [ ] Enforces code style
- [ ] Warns about a potential error
- [X] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```lang-typescript
import { createComponent, reactive, ref } from '@vue/composition-api';

export default createComponent({
	setup(){
		
		const first = ref(true);
		const second = reactive({a: 8, b: 9});
		
		const aMethod = () => {
			return second.a * 2;
		};
		
	}
```
Variables `first` and `aMethod` return an `no-unused-vars` error.
Do you think is possible to disable the rule for this type of variables / methods

**Comment by ota-meshi** [comment]
Thank you for posting this issue.

But I didn't understand.
Why do you need to disable it?

**Comment by ndr47** [comment]
Thanks for answering. I make a better example, where I am using the variables in the template.

```vue
<template>
	<div>
		<p>{{ first }}</p>
		<p>{{ aMethod }}</p>
		<button @click="first = !first">CLICK</button>
	</div>
</template>

<script lang="ts">
import { createComponent, reactive, ref } from '@vue/composition-api';

export default createComponent({
	setup(){
		
		const first = ref(true);
		const second = reactive({a: 8, b: 9});
		
		const aMethod = () => {
			return second.a * 2;
		};
		
	}
});

</script>
```

Then even if I am using the variables in the template I receive the error.

**Comment by ota-meshi** [comment]
I think I need a return statement, is that not?

```diff
	setup(){
		
		const first = ref(true);
		const second = reactive({a: 8, b: 9});
		
		const aMethod = () => {
			return second.a * 2;
		};
+		return { first, aMethod }
	}
```

I don't know how to create a development environment to work it.
Could you share your simple repository?

**Comment by ndr47** [comment]
I think you are right. Sorry about that. I forgot to return them.
